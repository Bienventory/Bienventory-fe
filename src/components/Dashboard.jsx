import React from 'react';
import Box from '@mui/material/Box';
// import { Link } from '@mui/material';
import { Link } from 'react-router-dom';
import { DataGrid } from '@mui/x-data-grid';

const dataArray = [
  {
    id: 1,
    item_name: 'potatoes',
    description: 'starch, staple',
    total_on_hand: 10,
    unit_type: 'lbs',
    restock_level: 2,
  },
  {
    id: 2,
    item_name: 'carrots',
    description: 'orange, staple',
    total_on_hand: 7,
    unit_type: 'lbs',
    restock_level: 2,
  },
  {
    id: 3,
    item_name: 'eggs',
    description: 'protein',
    total_on_hand: 24,
    unit_type: 'count',
    restock_level: 6,
  }
];

const rowsArray = dataArray.map((item) => {
  return {
    id: item.id,
    col1: item.item_name,
    col2: item.total_on_hand,
    col3: item.unit_type,
  };
});

const rows = rowsArray;

const columns = [
  { field: 'id', hide: true },
  { field: 'col1',
    headerName: 'Item Name',
    renderCell: (params) => (
      <Link to={`/ingredient/${params.value}`}>{params.value}</Link>
    )
  },
  { field: 'col2', headerName: 'On Hand' },
  { field: 'col3', headerName: 'Units' },
];

export default function Dashboard() {
  return(
    <Box sx={{ height: 300, width: '100%' }}>
      <DataGrid rows={rows} columns={columns}/>
    </Box>
  );
}
