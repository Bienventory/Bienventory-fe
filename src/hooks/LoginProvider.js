import React, { createContext, useContext, useState } from 'react';
import PropTypes from 'prop-types';

export const Context = createContext();

//context: user authentication, inventory array, menu array

const LoginProvider = ({ children }) => {
  const [user, setUser] = useState('');
  const [loggedIn, setloggedIn] = useState(false);

  const logInUser = async (googleId) => {
    // const response = await fetch(`theapi.urrl.com///${googleId}`);
    // const userObj = await response.json();
    const userObj = { google_id: googleId, notifications: false };

    setUser(userObj);
    setloggedIn(true);
  };

  const logOutUser = () => {
    setUser('');
    setloggedIn(false);
  };

  return (
    <Context.Provider value={{ user, loggedIn, logInUser, logOutUser }}>
      {children}
    </Context.Provider>
  );
};

export const uselogInUser = () => {
  const { logInUser } = useContext(Context);
  return logInUser;
};

export const uselogOutUser = () => {
  const { logOutUser } = useContext(Context);
  return logOutUser;
};

export const useloggedIn = () => {
  const { loggedIn } = useContext(Context);
  return loggedIn;
};

export const useUser = () => {
  const { user } = useContext(Context);
  return user;
};

LoginProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default LoginProvider;
