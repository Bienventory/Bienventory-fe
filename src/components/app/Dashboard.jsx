/* eslint-disable max-len */
import React, { useEffect, useState } from 'react';
import Box from '@mui/material/Box';
// import { Link } from '@mui/material';
import { Link } from 'react-router-dom';
import { DataGrid } from '@mui/x-data-grid';
import { useUser } from '../../hooks/LoginProvider';

const apiUrl = 'https://bienventory.herokuapp.com/api/v1/inventory/';

// const rowsArray = dataArray.map((item) => {
//   return {
//     id: item.id,
//     col1: item.item_name,
//     col2: item.total_on_hand,
//     col3: item.unit_type,
//   };
// });

export default function Dashboard() {
  const user = useUser();
  const [userData, setUserData] = useState([]);

  useEffect(() => {
    const loadData = async () => {
      const dataArray = await fetch(`${apiUrl}`);
      const json = await dataArray.json();
      const userData = json.filter((item) => {
        return user.google_id === item.user_id;
      });
      setUserData(userData);
    };
    loadData();
  }, []);

  const rows = userData.map((item) => {
    return {
      id: item.id,
      col1: item.item_name,
      col2: item.total_on_hand,
      col3: item.unit_type,
    };
  });

  const columns = [
    {
      field: 'id',
      hide: true,
    },
    {
      field: 'col1',
      headerName: 'Item Name',
      renderCell: (params) => (
        <Link to={`/inventory/${params.row.id}`}>{params.value}</Link>
      ),
    },
    { field: 'col2', headerName: 'On Hand' },
    { field: 'col3', headerName: 'Units' },
  ];

  return (
    <Box sx={{ height: '90%', width: '100%' }}>
      <DataGrid sx={{ height: '100%', width: '100%' }} rows={rows} columns={columns} />
    </Box>
  );
}
