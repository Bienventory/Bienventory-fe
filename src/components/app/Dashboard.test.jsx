import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import Dashboard from './Dashboard';
import inventory from '../../fixtures/inventory.json';
import LoginProvider from '../../hooks/LoginProvider';

const server = setupServer(
  rest.get(
    'https://bienventory.herokuapp.com/api/v1/inventory/',
    (req, res, ctx) => {
      return res(ctx.json(inventory));
    }
  )
);

describe('Dashboard Component', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it.skip('displays the inventory on the dashboard', async () => {
    const { container } = render(
      <LoginProvider>
        <MemoryRouter>
          <Dashboard />
        </MemoryRouter>
      </LoginProvider>
    );

    const inventoryGrid = await screen.findAllByRole('row');

    expect(inventoryGrid).not.toBeEmptyDOMElement;
    expect(container).toMatchSnapshot();
  });
});
