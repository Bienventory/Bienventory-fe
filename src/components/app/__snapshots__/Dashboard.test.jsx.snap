// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component displays the inventory on the dashboard 1`] = `
<div>
  <div
    class="MuiBox-root css-omukb7"
  >
    <div
      aria-colcount="3"
      aria-multiselectable="false"
      aria-rowcount="0"
      class="MuiDataGrid-root MuiDataGrid-root"
      role="grid"
    >
      <div>
        <div />
      </div>
      <div
        class="MuiDataGrid-main"
      >
        <div
          class="MuiDataGrid-overlay"
          style="top: 56px;"
        >
          No rows
        </div>
        <div
          class="MuiDataGrid-columnsContainer"
          style="min-height: 56px; max-height: 56px; line-height: 56px;"
        >
          <div
            aria-rowindex="1"
            class="MuiDataGrid-columnHeaderWrapper scroll"
            role="row"
            style="transform: translate3d(0px, 0px, 0px); min-width: 300px;"
          >
            <div
              aria-colindex="1"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="col1"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="0"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                    Item Name
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-controls="mui-138356894"
                    aria-haspopup="true"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                    id="mui-973190682"
                    tabindex="-1"
                    title="Menu"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="2"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="col2"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                    On Hand
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-controls="mui-474261703"
                    aria-haspopup="true"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                    id="mui-978797257"
                    tabindex="-1"
                    title="Menu"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
            <div
              aria-colindex="3"
              class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
              data-field="col3"
              role="columnheader"
              style="width: 100px; min-width: 100px; max-width: 100px;"
              tabindex="-1"
            >
              <div
                class="MuiDataGrid-columnHeaderDraggableContainer"
                draggable="false"
              >
                <div
                  class="MuiDataGrid-columnHeaderTitleContainer"
                >
                  <div
                    aria-label=""
                    class="MuiDataGrid-columnHeaderTitle"
                    data-mui-internal-clone-element="true"
                  >
                    Units
                  </div>
                  <div
                    class="MuiDataGrid-iconButtonContainer"
                  >
                    <button
                      aria-label="Sort"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="-1"
                      title="Sort"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ArrowUpwardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-menuIcon"
                >
                  <button
                    aria-controls="mui-277029125"
                    aria-haspopup="true"
                    aria-label="Menu"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall MuiDataGrid-menuIconButton css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                    id="mui-880921084"
                    tabindex="-1"
                    title="Menu"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="TripleDotsVerticalIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiDataGrid-columnSeparator"
                style="min-height: 56px; opacity: 1;"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                  data-testid="SeparatorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 19V5h2v14z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            class="MuiDataGrid-windowContainer"
            style="width: 0px; height: 0px;"
          >
            <div
              class="MuiDataGrid-window"
              style="top: 56px; overflow-y: auto;"
            >
              <div
                class="MuiDataGrid-dataContainer"
                style="min-width: 300px; min-height: 1px;"
              >
                <div
                  class="MuiDataGrid-viewport"
                  style="min-width: 0; max-width: 0; max-height: 0;"
                >
                  <div
                    class="MuiDataGrid-renderingZone"
                    style="max-height: 1px; width: 300px; transform: translate3d(0px, 0px, 0px);"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Mui-resizeTriggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1px; height: 1px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
      <div>
        <div
          class="MuiDataGrid-footerContainer"
        >
          <div />
          <div
            class="MuiTablePagination-root css-jtlhu6-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-11bfvty-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="makeStyles-selectLabel-1 MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                id="mui-388869560"
              >
                Rows per page:
              </p>
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary makeStyles-input-3 MuiTablePagination-input MuiSelect-root css-16c50h-MuiInputBase-root-MuiTablePagination-select"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="mui-388869560 mui-829618514"
                  class="MuiTablePagination-select MuiSelect-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                  id="mui-829618514"
                  role="button"
                  tabindex="0"
                >
                  100
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  tabindex="-1"
                  value="100"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiTablePagination-selectIcon MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                0-0 of 0
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
